gateway-server/src/main/java/org/apache/knox/gateway/websockets/ProxyWebSocketAdapter.java

```diff
@@ -57,6 +61,12 @@
 
   private ExecutorService pool;
 
+  /* Message buffer for holding data frames temporarily in memory till connection is setup.
+   Keeping the max size of the buffer as 100 messages for now. */
+  private List<String> messageBuffer;
+  private static final int MAX_BUFFER_MESSAGE_COUNT = 100;
```

Comment:
You may consider adding a new configuration - with a default value of 100 - in gateway-site.xml instead og hard-coding this limitation here.

Reply:
I am not sure if this should be added as a configuration in gateway-site.xml. The change is fixing a race condition which is very short lived (few milliseconds).

Reply:
+1 to @smolnar82 comment.

Reply:
Made changes to expose the buffer size as a new configuration. Please check.